torch/testing/_internal/common_methods_invocations.py

```diff
@@ -349,6 +349,70 @@ def wrapped_fn(x):
 
     return wrapped_fn
 
+def np_binary_ufunc_type_promotion_wrapper(fn):
+    # Wrapper that passes PyTorch's default scalar
+    #   type as an argument to the wrapped NumPy
+    #   binary ufunc when given an integer input .
+    #   This mimicks PyTorch's integer->floating point
+    #   type promotion.
+    #
+    # This is necessary when NumPy promotes
+    #   integer types to double, since PyTorch promotes
+    #   integer types to the default scalar type.
+    #
+    # Besides, the table below shows the difference between
+    #   PyTorch and NumPy on the type promotion for binary
+    #   functions
+    #      inputs         NumPy     PyTorch
+    #    bool  bool       vary        bool
+    #   int16 float16    float32    float16
+    #   int32 float16    float64    float16
+    #   int64 float16    float64    float16
+    #   int32 float32    float64    float32
+    #   int64 float32    float64    float32
+    #
+    # Another special case only on Windows platform due to
+    # torch tensor wrapper for integer scalar is int64, but
+    # numpy wrapper for integer scalar is int32.
```

Comment:
are you sure this is correct and it's not based on the value of the scalar?

Reply:
Here is my experiment on the Windows machine on CI and another Windows machine:
```python
>>> a = 1
>>> torch.tensor(a).dtype
torch.int64
>>> np.array(a).dtype
dtype('int32')
```

Reply:
At least for `fmod` and `remainder`.
- Windows
```python
>>> x = np.array(True)
>>> np.fmod(x, 111).dtype
dtype('int32')
```
- MacOS/Linux
```python
>>> x = np.array(True)
>>> np.fmod(x, 111).dtype
dtype('int64')
```

Reply:
Looks to me like you're both correct. NumPy converts Python integers to its default integer type, which is the equivalent of C long and therefore platform-dependent. On most platforms the result will be `int64`, on Windows and on (e.g.) 32-bit Linux it'll be `int32`.

In addition, NumPy does value-based casting, so if the Python integer doesn't fit in 32 bits it should choose `int64`. I don't have a Windows machine at hand, but I'm pretty sure this will be the result on Windows:
```
>>> np.array(np.iinfo(np.int32).max * 10).dtype
dtype('int64')
```